### ISSUE TITLE
Assertion CBC_FUNCTION_GET_TYPE (frame_ctx_p->bytecode_header_p->status_flags) == CBC_FUNCTION_ASYNC in opfunc_async_create_and_await

### ISSUE THREAD

**Issue by ossy-szeged** [jerryscript-project/jerryscript#4135]
###### JerryScript revision
0c154306a87492f255edae6f483a768f128c2c09 
(latest master - 2020.08.11.)

###### Build platform
Ubuntu 18.04.5 LTS (Linux 4.15.0-112-generic x86_64)

###### Build steps
tools/build.py --debug --logging=on --error-messages=on --line-info=on

###### Test case
found by test262-esnext language/module-code/top-level-await/... tests

minimized test case
```
function fn() { return function() {}; }
export class C extends fn(await []) {};
```

###### Output
```
ICE: Assertion 'CBC_FUNCTION_GET_TYPE (frame_ctx_p->bytecode_header_p->status_flags) == CBC_FUNCTION_ASYNC || CBC_FUNCTION_GET_TYPE (frame_ctx_p->bytecode_header_p->status_flags) == CBC_FUNCTION_ASYNC_ARROW' failed at /home/oszi/jerryscript/jerry-core/vm/opcodes.c(opfunc_async_create_and_await):914.
Error: ERR_FAILED_INTERNAL_ASSERTION
Aborted (core dumped)
```

###### Expected behavior
not assert, but SyntaxError, because top-level-await isn't supported yet

**Comment by ossy-szeged** [comment]
Still valid on latest master ( 0ffe1665bd0ee457dd4c1083811e1592a5e2908d ).

**Comment by zherczeg** [comment]
I get a SyntaxError now, maybe we can close this?

**Comment by rerobika** [comment]
Closed via #4266 .
